src/orca-jedi/increment/Increment.cc

```diff
@@ -84,20 +85,28 @@ Increment::Increment(const Increment & other, const bool copy)
 
   setupIncrementFields();
 
+  auto ghost = atlas::array::make_view<int32_t, 1>(
+      geom_->mesh().nodes().ghost());
   if (copy) {
     for (size_t i=0; i < vars_.size(); ++i) {
       // copy variable from _Fields to new field set
       atlas::Field field = other.incrementFields_[i];
       oops::Log::debug() << "Copying increment field " << field.name() << std::endl;
-      incrementFields_->add(field);
+      auto field_view = atlas::array::make_view<double, 2>(incrementFields_[i]);
+      auto field_view_other = atlas::array::make_view<double, 2>(field);
+      for (atlas::idx_t j = 0; j < field_view.shape(0); ++j) {
+        for (atlas::idx_t k = 0; k < field_view.shape(1); ++k) {
+          if (!ghost(j)) field_view(j, k) = field_view_other(j, k);
```

Comment:
Is copying only non-ghost points the correct behaviour? Presumably in the old version the whole thing would have been copied including ghosts?

Reply:
I made the change to copy across all points and it seemed to run OK.